sktalk/corpus/parsing/parser.py

```diff
@@ -8,10 +7,10 @@ class InputFile(abc.ABC):
     def __init__(self, path: str) -> None:
         self._path = path
         self._metadata = {"source": path}
+        self._utterances = []
 
-    @abc.abstractmethod
-    def parse(self) -> Conversation:
-        return NotImplemented
+    def parse(self):
```

Comment:
I would suggest adding a type hint to indicate the return values:

```suggestion
    def parse(self) -> tuple["Utterance", dict]:
```
Note the quotation marks so that you don't have to import `Utterance` just for the purpose of type checking. However, the type checker might complain so you might have to import it specifically for the type checking using the [`TYPE_CHECKING` constant](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING).